libvast/src/segment_store.cpp

```diff
@@ -357,6 +356,9 @@ caf::error segment_store::register_segment(const path& filename) {
   if (!chk)
     return make_error(ec::filesystem_error, "failed to mmap chunk", filename);
   auto s = fbs::GetSegment(chk->data());
+  auto verifier = fbs::make_verifier(chk);
+  if (!fbs::VerifySegmentBuffer(verifier))
+    return make_error(ec::format_error, "flatbuffer integrity check failed");
```

Comment:
Nit: the verification call to `fbs::GetSegment()` can be done after the verification.

Reply:
I'm also not sure this is a good idea. Consider the case of a large segment store (think of several TB): This verification means that we have to calculate the checksum for every segment during startup, which will probably take a long time.

Reply:
> Nit: the call to `fbs::GetSegment()` can be done after the verification.

Okay, I'll move it over.

> I'm also not sure this is a good idea. Consider the case of a large segment store (think of several TB): This verification means that we have to calculate the checksum for every segment during startup, which will probably take a long time.

It's not a checksum. It's just a verification that all offsets are within bounds and don't yield segfaults. It's not a full scan over the data. This is how the layout looks like:

[![FBPerson01](https://user-images.githubusercontent.com/53797/76536841-2c96ee80-647d-11ea-8646-22b1ed251abd.png)](https://github.com/mzaks/FlatBuffersSwift/wiki/FlatBuffers-Explained)

It's my understanding that verification only requires the initial table state.

Reply:
I see, this is good to go then.